File: Assignment 1/answers/person.js
const Uuid = require('uuid');
const faker = require('faker');

class Person {


    constructor(name, age, email, gender = Gender.Male) {
        this.name = name;
        this.age = age;
        this.gender = gender;
        this.email=email;
        this.id = Uuid.v4();
    }

    // Method to convert object to json
    toJson() {
        return {
            name: this.name,
            age: this.age,
            id: this.id,
            gender: this.gender,
            email:this.email,
        }
    }


    // a static method that creates a fake person
    static fake() {
        return new Person(faker.name.findName(), faker.datatype.number(100),faker.internet.email());
    }

}

const Gender = {
    Male: "Male",
    Female: "Female"
}

module.exports = Person;

File: Assignment 2/answers/frontend_flutter_src/flutter_application_1/lib/models/person.dart
import 'package:json_annotation/json_annotation.dart';

part 'person.g.dart';

@JsonSerializable()
class Person {
  String id;
  String name;
  int age;
  String email;
  Gender gender;

  Person(
      {required this.id,
      required this.name,
      required this.age,
      required this.email,
      required this.gender});

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);
  Map<String, dynamic> toJson() => _$PersonToJson(this);

  @override
  @override
  String toString() =>
      'Person{name: $name}, age:$age, email:$email, gender:${gender.name}';
}

enum Gender { Male, Female }
